crates/rpc/rpc-eth-api/src/helpers/estimate.rs

```diff
@@ -128,7 +141,27 @@ pub trait EstimateCall: Call {
                 if err.is_gas_too_high() &&
                     (tx_request_gas_limit.is_some() || tx_request_gas_price.is_some()) =>
             {
-                return Err(self.map_out_of_gas_err(block_env_gas_limit, evm_env, tx_env, &mut db))
+                let req_gas_limit = tx_env.gas_limit();
+                let mut retry_tx_env = tx_env.clone();
+                retry_tx_env.set_gas_limit(block_env_gas_limit);
+
+                let retry_res = evm.transact(retry_tx_env).map_err(Self::Error::from_evm_err)?;
+
+                return match retry_res.result {
+                    ExecutionResult::Success { .. } => {
+                        // transaction succeeded by manually increasing the gas limit to
+                        // highest, which means the caller lacks funds to pay for the tx
+                        Err(RpcInvalidTransactionError::BasicOutOfGas(req_gas_limit).into_eth_err())
+                    }
+                    ExecutionResult::Revert { output, .. } => {
+                        // reverted again after bumping the limit
+                        Err(RpcInvalidTransactionError::Revert(RevertError::new(output))
+                            .into_eth_err())
+                    }
+                    ExecutionResult::Halt { reason, .. } => {
+                        Err(Self::Error::from_evm_halt(reason, req_gas_limit))
+                    }
+                };
```

Comment:
why do we need to do this here now?

we didn't transact in this case previously